src/boot/bootctl-uki.c

```diff
@@ -109,3 +111,74 @@ int verb_kernel_identify(int argc, char *argv[], void *userdata) {
         puts("unknown");
         return EXIT_SUCCESS;
 }
+
+static int read_pe_section(FILE *uki, struct PeSectionHeader *section,
+                           void **ret, int *ret_n) {
+        _cleanup_free_ void *data = NULL;
+        uint64_t soff;
+        uint32_t size;
+        uint32_t bytes;
+        int rc;
+
+        soff = le32toh(section->PointerToRawData);
+        size = le32toh(section->VirtualSize);
+
+        rc = fseek(uki, soff, SEEK_SET);
+        if (rc < 0)
+                return log_error_errno(errno, "seek to PE section");
+
+        data = malloc(size);
+        bytes = fread(data, 1, size, uki);
+        if (bytes != size)
+                return log_error_errno(SYNTHETIC_ERRNO(EIO), "PE section read error");
+
+        *ret = TAKE_PTR(data);
+        *ret_n = size;
+        return 0;
+}
+
+static void inspect_uki(FILE *uki, struct PeSectionHeader *sections, uint16_t scount) {
+        _cleanup_free_ char *cmdline = NULL;
+        _cleanup_free_ char *uname = NULL;
+        int cmdline_len, uname_len;
+        uint16_t index;
```

Comment:
index() is also the name of a libc function. Maybe call this "idx" to avoid the name clash. (old gcc used to warn about this, nowadays they don't anymore, but i'd still avoid it). maybe just name this "idx"?